### PR TITLE
feat: Print out the Global Config object from within the BUILD file

### PR DESCRIPTION
**Pull request by PeterBocan**
This change allows printing out the builtin `CONFIG` file from the BUILD files.

### CODE REVIEW

File: src/parse/asp/objects.go [hunk 1]
```diff
@@ -822,7 +822,7 @@ type pyConfig struct {
 
 func (c *pyConfig) MarshalJSON() ([]byte, error) {
 	if c.overlay == nil {
-		return json.Marshal(c.overlay)
```
Comment (PeterBocan) [inline src/parse/asp/objects.go:825]: this was possibly a bug? Looks a bit sus...

File: src/parse/asp/objects.go [hunk 2]
```diff
@@ -835,7 +835,11 @@ func (c *pyConfig) MarshalJSON() ([]byte, error) {
 }
 
 func (c *pyConfig) String() string {
-	return "<global config object>"
+	marshalledConfig, err := c.MarshalJSON()
+	if err != nil {
+		panic(fmt.Sprintf("failed to marshal object: %s", err))
+	}
+	return fmt.Sprintf("GlobalConfig%s", string(marshalledConfig))
 }
```
Comment (Tatskaari) [inline src/parse/asp/objects.go:843]: This is quite different to `pyDict` which uses this formatting:
```
func (d pyDict) String() string {
	var b strings.Builder
	b.WriteByte('{')
	started := false
	for _, k := range d.Keys() {
		if started {
			b.WriteString(", ")
		}
		started = true
		b.WriteByte('"')
		b.WriteString(k)
		b.WriteString(`": `)
		b.WriteString(d[k].String())
	}
	b.WriteByte('}')
	return b.String()
}
```

I feel like these two should behave the same.

Comment (PeterBocan) [reply src/parse/asp/objects.go]: Fair point! Thanks, I have addressed that.

File: src/parse/asp/objects.go (hunk 1 above)
Comment (Tatskaari) [reply src/parse/asp/objects.go:825]: Yeah almost certainly. nice spot!

File: src/parse/asp/objects.go [hunk 3]
```diff
@@ -822,20 +822,40 @@ type pyConfig struct {
...
+			b.WriteString(", ")
+		}
+		started = true
+		b.WriteByte('"')
+		b.WriteString(key)
+		b.WriteString(`": `)
+		b.WriteString(config[key].String())
+	}
+	b.WriteByte('}')
+	return b.String()
```
Comment (Tatskaari) [inline src/parse/asp/objects.go:858]: Can't we just do `c.Config().String()`, using the `pyDict` implementation?

File: src/parse/asp/objects.go [hunk 4]
```diff
@@ -822,20 +822,40 @@ type pyConfig struct {
...
 func (c *pyConfig) MarshalJSON() ([]byte, error) {
 	if c.overlay == nil {
-		return json.Marshal(c.overlay)
+		return json.Marshal(c.base.dict)
 	}
+
+	return json.Marshal(c.Config())
+}
+
+func (c *pyConfig) Config() pyDict {
```
Comment (Tatskaari) [inline src/parse/asp/objects.go:831]: (nit: needs a doc comment)

but does this need to be public? Maybe make it private and re-name it `toPyDict()`?

**Pull request merged**